cclib/method/ddec.py

```diff
@@ -0,0 +1,170 @@
+        return "DDEC6({})".format(self.data)
+
+    def _check_required_attributes(self):
+        super(DDEC6, self)._check_required_attributes()
+
+    def _read_proatom(self, directory, atom_num, charge):
+        """Return a list containing proatom reference densities."""
+        # TODO: Treat calculations with psuedopotentials
+        # File name format:
+        #   ** Chargemol **
+        #       c2_[atom number]_[nuclear charge]_[electron count]_[cutoff radius]_[# shells]
+        #   ** Horton **
+        #       atoms.h5
+        # File format:
+        #   Starting from line 13, each line contains the charge densities for each shell
+        chargemol_path = os.path.join(
+            directory,
+            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge),
+        )
+        horton_path = os.path.join(directory, "atoms.h5")
+
+        if os.path.isfile(chargemol_path):
+            # Use chargemol proatom densities
+            # Each shell is 5.600570811644798 angstroms apart (uniform).
+            # *scalefactor* = 10.58354497764173 in module_global_parameter.f08
+            density = numpy.loadtxt(path, skiprows=12, dtype=float)
+            radiusgrid = numpy.arange(1, len(density + 1)) * 5.600570811644798
+
+        elif os.path.isfile(horton_path):
+            # Use horton proatom densities
+            _found_h5py = find_package("h5py")
+            assert _found_h5py, ("h5py is needed to read in proatom densities from horton.")
+
+            import h5py
+
+            with h5py.File(horton_path, "r") as proatomdb:
+                keystring = "Z={}_Q={:+d}".format(atom_num, charge)
+
+                # gridspec is specification of integration grid for proatom densities in horton.
+                # Example -- ['PowerRTransform', '1.1774580743206259e-07', '20.140888089596444', '41']
+                #   is constructed using PowerRTransform grid
+                #   with rmin = 1.1774580743206259e-07
+                #        rmax = 20.140888089596444
+                #   and  ngrid = 41
+                # PowerRTransform is default in horton-atomdb.py.
+                gridtype, gridmin, gridmax, gridn = proatomdb[keystring].attrs["rtransform"].split()
+                gridmin = convertor(float(gridmin), "bohr", "Angstrom")
+                gridmax = convertor(float(gridmax), "bohr", "Angstrom")
+                gridn = int(gridn)
+                # Convert byte to string in Python3
+                if sys.version[0] == '3':
+                    gridtype = gridtype.decode('UTF-8')
+
+                # First verify that it is one of recognized grids
+                assert gridtype in ["LinearRTransform", "ExpRTransform", "PowerRTransform"], "Grid type not recognized."
+
+                if gridtype == "LinearRTransform":
+                    # Linear transformation. r(t) = rmin + t*(rmax - rmin)/(npoint - 1)
+                    gridcoeff = (gridmax - gridmin) / (gridn - 1)
```

Comment:
Are these guaranteed to be floats? If not, can be problematic for Python2.

Reply:
To my knowledge, these would be guaranteed to be floats due to lines 115-116

```
gridmin = convertor(float(gridmin), "bohr", "Angstrom")
gridmax = convertor(float(gridmax), "bohr", "Angstrom")
```

but if I missed anything, please let me know.